Client/ClientInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Client;
+
+interface ClientInterface
```

Comment:
if you want to avoid this code, you could use the [HttpMethodsClient](http://docs.php-http.org/en/latest/components/client-common.html#httpmethodsclient) which would allow $client->get, $client->post and so on in addition to ->send($method, ...). 

there is the php-http/guzzle-adapter to wrap current guzzle versions into the psr18 client so it can be used with the HttpMethodsClient.

if you want your own interface, i would recommend calling the method `send` rather than `sendRequest` - we called the method sendRequest in psr18 because it accepts a request object rather than strings.